src/ardrive.ts

```diff
@@ -240,6 +241,13 @@ export class ArDrive extends ArDriveAnonymous {
 			throw new Error(errorMessage.cannotMoveIntoSamePlace('File', newParentFolderId));
 		}
 
+		// Assert that there are no duplicate names in the destination folder
+		const entityNamesInParentFolder = await this.arFsDao.getPublicEntityNamesInFolder(newParentFolderId);
+		if (entityNamesInParentFolder.includes(originalFileMetaData.name)) {
+			// TODO: Add optional interactive prompt to resolve name conflicts in ticket PE-599
+			throw new Error(errorMessage.entityNameExists);
+		}
+
```

Comment:
I think we could consolidate this repeated code by extracting these new **async** functions:
```ts
assertUniqueNameInFolder(names: string[], name: string): Promise<void>;
assertNoOverridingFile(files: FileNameAndID[], name: string): Promise<void>;
assertNoOverridingFolder(folders: FolderNameAndID[], name: string): Promise<void>;
```

Reply:
Agreed, and not just the ones from this PR -- we could consolidate a lot of the business logic errors in this class.

I've added a tech_debt ticket for this in phase 1.5: https://ardrive.atlassian.net/browse/PE-609